webapp/channels/src/components/code_preview.tsx

```diff
@@ -20,130 +20,115 @@ type Props = {
-            }
+        if (!usedLanguage || fileInfo.size > Constants.CODE_PREVIEW_MAX_FILE_SIZE) {
+            setCodeInfo({...codeInfo, code: '', lang: ''});
 
-            return {
-                code: '',
-                lang: usedLanguage,
-                loading: true,
-                prevFileUrl: props.fileUrl,
-            };
-        }
-        return null;
-    }
+            setStatus({loading: false, success: false});
+        } else {
+            setCodeInfo({...codeInfo, code: '', lang: usedLanguage});
 
-    componentDidUpdate(prevProps: Props) {
-        if (this.props.fileUrl !== prevProps.fileUrl) {
-            this.getCode();
+            setStatus({...status, loading: true});
         }
+
+        setPrevFileUrl(fileUrl);
     }
 
-    getCode = async () => {
-        if (!this.state.lang || this.props.fileInfo.size > Constants.CODE_PREVIEW_MAX_FILE_SIZE) {
-            return;
-        }
-        try {
-            const data = await fetch(this.props.fileUrl);
-            const text = await data.text();
-            this.handleReceivedCode(text);
-        } catch (e) {
-            this.handleReceivedError();
-        }
-    };
+    useEffect(() => {
+        const handleReceivedCode = async (data: string | Node) => {
+            let code = data as string;
+            const Data = data as Node;
 
-    handleReceivedCode = async (data: string | Node) => {
-        let code = data as string;
-        const Data = data as Node;
-        if (Data.nodeName === '#document') {
-            code = new XMLSerializer().serializeToString(Data);
-        }
-        this.props.getContent?.(code);
-        this.setState({
-            code,
-            highlighted: await SyntaxHighlighting.highlight(this.state.lang, code),
-            loading: false,
-            success: true,
-        });
-    };
-
-    handleReceivedError = () => {
-        this.setState({loading: false, success: false});
-    };
-
-    static supports(fileInfo: FileInfo | LinkInfo) {
-        return Boolean(SyntaxHighlighting.getLanguageFromFileExtension(fileInfo.extension));
-    }
+            if (Data.nodeName === '#document') {
+                code = new XMLSerializer().serializeToString(Data);
+            }
 
-    render() {
-        if (this.state.loading) {
-            return (
-                <div className='view-image__loading'>
-                    <LoadingSpinner/>
-                </div>
-            );
-        }
+            getContent?.(code);
 
-        if (!this.state.success) {
-            return (
-                <FileInfoPreview
-                    fileInfo={this.props.fileInfo}
-                    fileUrl={this.props.fileUrl}
-                />
-            );
-        }
+            setCodeInfo({
+                ...codeInfo,
+                code,
+                highlighted: await SyntaxHighlighting.highlight(codeInfo.lang, code),
+            });
+
+            setStatus({loading: false, success: true});
+        };
```

Comment:
Use the function version of set: `setCodeInfo((prev) => {...})`. That way this effect doesn't depend that much on `codeInfo`, that may change for many reasons.

Reply:
The [latest changes](https://github.com/mattermost/mattermost/pull/33302#pullrequestreview-3087787150) have been approved, so I'm marking this conversation as resolved.